packages/flutter/lib/src/material/page.dart

```diff
@@ -205,6 +202,8 @@ class MaterialPageRoute<T> extends PageRoute<T> {
     backGestureController?.dispose();
   }
 
+  bool get _useCupertinoTransition => false;
```

Comment:
why put the logic in the base class configured with a flag instead of just putting the logic in the subclass?